src/TestableIO.System.IO.Abstractions.TestingHelpers/MockFileVersionInfo.cs

```diff
@@ -0,0 +1,207 @@
+﻿using System.Diagnostics;
+using System.Runtime.InteropServices;
+using System.Text;
+
+namespace System.IO.Abstractions.TestingHelpers
+{
+    /// <inheritdoc />
+#if FEATURE_SERIALIZABLE
+    [Serializable]
+#endif
+    public class MockFileVersionInfo : FileVersionInfoBase
+    {
+        /// <inheritdoc />
+        public MockFileVersionInfo(
```

Comment:
Why not set a default value for the optional arguments?
Then I could also specify, e.g.
```csharp
new MockFileVersionInfo("myFile.dll", companyName: "My company");
```
when I am only interested in a specific part of the `IFileVersionInfo`

This should make the other constructors obsolete...

Reply:
When I tried that, your "Codacy static code analysis" workflow failed. [Link](https://github.com/TestableIO/System.IO.Abstractions/runs/34494707382)

But at that time, all of the parameters were optional. Maybe thats what caused the error.
I will try it again, but this time, the file name will be mandatory.

Reply:
Well.. I tried and it fails again with "Use the overloading mechanism instead of the optional parameters."

Reply:
I will have a look...

Reply:
I deactivated this rule in Codacy and it is now working...

Reply:
Cool. Thanks.